pkg/apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -100,3 +102,20 @@ type SubmarinerList struct {
 func init() {
 	SchemeBuilder.Register(&Submariner{}, &SubmarinerList{})
 }
+
+func (submariner *Submariner) SetDefaults() {
+
+	if submariner.Spec.Repository == "" {
+		// An empty field is converted to the default upstream submariner repository where all images live
+		submariner.Spec.Repository = versions.DefaultSubmarinerRepo
+	}
+
+	if submariner.Spec.Version == "" {
+		submariner.Spec.Version = versions.DefaultSubmarinerVersion
+	}
+
+	if submariner.Spec.ColorCodes == "" {
+		submariner.Spec.ColorCodes = "blue"
```

Comment:
nit: default colorCode of "blue" is used at couple of places in the code. Shall we create a const for that?

Reply:
Makes sense, I'll send a separate PR for that one.